### PR TITLE
Java 21 and Springboot 3 upgrade work

### PR DESCRIPTION
**Pull request by hwatkins-ch**

### CODE REVIEW

**Review by dhawes-ch — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -444,6 +416,12 @@
                     <container>
                         <expandClasspathDependencies>true</expandClasspathDependencies>
                     </container>
+                    <from>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
+                    </from>
+                    <to>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/local/psc-statement-delta-consumer:latest</image>
```
Comment (foman-ch) [inline pom.xml:423]: same question as before for this image

**Comment by foman-ch** [comment]
Looks like this uses a salt that contains a £ symbol. It might be worth adding some temporary logging to check that value is being injected correctly into the service. 

I had a bug in overseas-entities related to the same (the fix to the DockerFile is included in the comments):
https://companieshouse.atlassian.net/browse/UAR-1594?focusedCommentId=258367&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-258367

To fix it locally add these lines to the docker-compose:
      - LANG=C.UTF-8
      - LC_ALL=C.UTF-8

**Comment by hwatkins-ch** [comment]
> Looks like this uses a salt that contains a £ symbol. It might be worth adding some temporary logging to check that value is being injected correctly into the service.
> 
> I had a bug in overseas-entities related to the same (the fix to the DockerFile is included in the comments): https://companieshouse.atlassian.net/browse/UAR-1594?focusedCommentId=258367&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-258367
> 
> To fix it locally add these lines to the docker-compose: - LANG=C.UTF-8 - LC_ALL=C.UTF-8

I

**Review by foman-ch — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (hwatkins-ch) [reply pom.xml:423]: Just checking with Abi

**Comment by hwatkins-ch** [comment]
> Looks like this uses a salt that contains a £ symbol. It might be worth adding some temporary logging to check that value is being injected correctly into the service.
> 
> I had a bug in overseas-entities related to the same (the fix to the DockerFile is included in the comments): https://companieshouse.atlassian.net/browse/UAR-1594?focusedCommentId=258367&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-258367
> 
> To fix it locally add these lines to the docker-compose: - LANG=C.UTF-8 - LC_ALL=C.UTF-8

I will get Nicole to include this change to the dockerfile in the ECS config on the migration branch, since it is probably required as its the same scenario as you saw in overseas-entity

**Review by akunati-ch — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (hwatkins-ch) [reply pom.xml]: Using old image reference until staging deployment done

**Pull request merged**